src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets

```diff
@@ -439,7 +439,7 @@ The .NET Foundation licenses this file to you under the MIT license.
   </Target>
 
   <Target Name="CompileSingleWasmObject">
-    <Exec Command="$(ExecCommand)" />
+    <Exec Command="$(ExecCommand)" EnvironmentVariables="@(EmscriptenEnvVars)" />
```

Comment:
What are these environment variables expected to affect?

This step is compiling bitcode files produces by ILC into object files. In the fullness of time, this step will be folded into ILC itself.

Reply:
We have a different `.emscripten` file. It relies on env vars to find other tools. We also provide a path to immutable prebuilt cache.

Reply:
I have added a preview of upstream changes to the PR description